cmd/gen-key/ecdsa.go

```diff
@@ -94,107 +84,52 @@ func ecArgs(label string, curve *elliptic.CurveParams, compatMode bool, keyID []
 
 // ecPub extracts the generated public key, specified by the provided object
 // handle, and constructs an ecdsa.PublicKey. It also checks that the key is of
-// the correct curve type. For devices that implement a pre-2.11 version of the
-// PKCS#11 specification compatMode should be true.
-func ecPub(ctx PKCtx, session pkcs11.SessionHandle, object pkcs11.ObjectHandle, expectedCurve *elliptic.CurveParams, compatMode bool) (*ecdsa.PublicKey, error) {
-	var paramType uint
-	if compatMode {
-		paramType = pkcs11.CKA_ECDSA_PARAMS
-	} else {
-		paramType = pkcs11.CKA_EC_PARAMS
-	}
-	// Retrieve the curve and public point for the generated public key
-	attrs, err := ctx.GetAttributeValue(session, object, []*pkcs11.Attribute{
-		pkcs11.NewAttribute(paramType, nil),
-		pkcs11.NewAttribute(pkcs11.CKA_EC_POINT, nil),
-	})
+// the correct curve type.
+func ecPub(ctx pkcs11helpers.PKCtx, session pkcs11.SessionHandle, object pkcs11.ObjectHandle, expectedCurve *elliptic.CurveParams) (*ecdsa.PublicKey, error) {
```

Comment:
As long as you're touching this line, let's split it into one line per argument.